pygsp/graphs/nngraphs/nngraph.py

```diff
@@ -1,199 +1,560 @@
+        * ``'minkowski'`` generalizes the above and defines distances as
+          :math:`d(v_i, v_j) = \| x_i - x_j \|_p`
+          where :math:`p` is the ``order`` of the norm.
+        * ``'max_dist'`` defines pairwise distances as
+          :math:`d(v_i, v_j) = \| x_i - x_j \|_\infty = \max(x_i - x_j)`, where
+          the maximum is taken over the elements of the vector.
+
+        More metrics may be supported for some backends.
+        Please refer to the documentation of the chosen backend.
     order : float, optional
-        Only used if dist_type is 'minkowski'; represents the order of the
-        Minkowski distance. (default is 0)
+        The order of the Minkowski distance for ``metric='minkowski'``.
+    kind : {'knn', 'radius'}, optional
+        Kind of nearest neighbor graph to create. Either ``'knn'`` for
+        k-nearest neighbors or ``'radius'`` for epsilon-nearest neighbors.
+    k : int, optional
+        Number of neighbors considered when building a k-NN graph with
+        ``type='knn'``.
+    radius : float or {'estimate', 'estimate-knn'}, optional
+        Radius of the ball when building a radius graph with ``type='radius'``.
+        It is hard to set an optimal radius. If too small, some vertices won't
+        be connected to any other vertex. If too high, vertices will be
+        connected to many other vertices and the graph won't be sparse (high
+        average degree).  If no good radius is known a priori, we can estimate
+        one. ``'estimate'`` sets the radius as the expected average distance
+        between vertices for a uniform sampling of the ambient space.
+        ``'estimate-knn'`` first builds a knn graph and sets the radius to the
+        average distance. ``'estimate-knn'`` usually gives a better estimation
+        but is more costly. ``'estimate'`` can be better in low dimension.
+    kernel : string or function
+        The function :math:`k` that transforms a distance to a similarity.
+        The following kernels are pre-defined.
+
+        * ``'gaussian'`` defines the Gaussian, also known as the radial basis
+          function (RBF), kernel :math:`k(d) = \exp(-\log(2) d^2)`.
+        * ``'exponential'`` defines the kernel :math:`k(d) = \exp(-\log(2) d)`.
+        * ``'rectangular'`` returns 1 if :math:`d < 1` and 0 otherwise.
+        * ``'triangular'`` defines the kernel :math:`k(d) = \max(1 - d/2, 0)`.
+        * Other kernels are ``'tricube'``, ``'triweight'``, ``'quartic'``,
+          ``'epanechnikov'``, ``'logistic'``, and ``'sigmoid'``.
+          See `Wikipedia <https://en.wikipedia.org/wiki/Kernel_(statistics)>`_.
+
+        Another option is to pass a function that takes a vector of pairwise
+        distances and returns the similarities. All the predefined kernels
+        return a similarity of 0.5 when the distance is one.
+        An example of custom kernel is ``kernel=lambda d: d.min() / d``.
+    kernel_width : float, optional
```

Comment:
I think we should call this `kernel_scale` or even `scale` instead of `kernel_width`.

Reply:
Good point. I don't have strong feelings, but we should settle on a good name, to not have to change it later. I like `scale`, as it is a scaling of the distances, before sending them to the kernel. What's your reasoning?